### PR TITLE
Fix unit test execution on MSVC

### PR DESCRIPTION
**Pull request by dnachbaur**

### CODE REVIEW

File: tests/cpp/SocketTests.cpp [hunk 1]
```diff
@@ -50,47 +50,33 @@ namespace ut = boost::unit_test;
 
 BOOST_GLOBAL_FIXTURE( MinimalGlobalQtApp );
 
-BOOST_AUTO_TEST_CASE( testSocketConnectionValidWhenReturnedCorrectNetworkProtocolVersion )
+void testSocketConnect( const int32_t versionOffset )
 {
     QThread thread;
-    MockServer server;
-    server.moveToThread( &thread );
+    MockServer* server = new MockServer( NETWORK_PROTOCOL_VERSION + versionOffset );
```
Comment (rdumusc) [inline tests/cpp/SocketTests.cpp:56]: Any reason for making this a ptr? Looks equivalent to me, but slightly more complex with the added connection...

**Comment by rdumusc** [comment]
+1

File: tests/cpp/SocketTests.cpp (hunk 1 above)
Comment (dnachbaur) [inline tests/cpp/SocketTests.cpp:56]: I'm pretty sure for QObject::deleteLater the object cannot be on the stack.
It's not equivalent, as the deleteLater connection ensures that the object is destroyed in the correct thread. Before, the server object was moved to the worker thread, but destroyed in the main thread. Qt on Windows was complaining about that.

Comment (rdumusc) [inline tests/cpp/SocketTests.cpp:56]: OK, interesting..! The thread is joined at the end of the scope, so I didn't think about this...

**Pull request merged**